pkg/flowcli/services/keys.go

```diff
@@ -78,16 +82,45 @@ func (k *Keys) Generate(inputSeed string, signatureAlgo string) (crypto.PrivateK
 	return privateKey, nil
 }
 
-func (k *Keys) Decode(publicKey string) (*flow.AccountKey, error) {
+// Decode encoded public key using supported encoding type
+func (k *Keys) Decode(encoded string, encoding string) (*flow.AccountKey, error) {
+	if strings.ToLower(encoding) == PEM {
+		return decodePEM(encoded)
+	} else if strings.ToLower(encoding) == RLP {
+		return decodeRLP(encoded)
+	}
+
+	return nil, fmt.Errorf("encoding type not supported. Valid encodings: RLP, PEM")
+}
+
+func decodeRLP(publicKey string) (*flow.AccountKey, error) {
 	publicKeyBytes, err := hex.DecodeString(publicKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to decode public key: %v", err)
 	}
 
 	accountKey, err := flow.DecodeAccountKey(publicKeyBytes)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode private key bytes: %v", err)
+		return nil, fmt.Errorf("failed to decode: %v", err)
```

Comment:
should `%w` be used instead of `%v` for errors? https://blog.golang.org/go1.13-errors#TOC_3.3.

Reply:
Ah yep, it's best to use `%w` when wrapping errors like this.

Reply:
TIL